be/src/io/fs/hdfs_file_reader.cpp

```diff
@@ -40,6 +40,10 @@ HdfsFileReader::~HdfsFileReader() {
 }
 
 Status HdfsFileReader::close() {
+    if (_closed) {
+        return Status::OK();
+    }
+    std::lock_guard<std::mutex> lock(_lock);
```

Comment:
Is it necessary to make a hdfs reader thread safe?

Reply:
The reader inner `BufferedReader` should be thread safe, otherwise it will core dump.